packages/hash/frontend/src/blocks/page/Comments/placeholderPlugin.tsx

```diff
@@ -0,0 +1,50 @@
+import Typography from "@mui/material/Typography";
+import { Schema } from "prosemirror-model";
+import { Plugin } from "prosemirror-state";
+import { Decoration, DecorationSet } from "prosemirror-view";
+import { RenderPortal } from "../usePortals";
+
+export const placeholderPlugin = (renderPortal: RenderPortal, text: string) =>
+  new Plugin<Schema, Schema>({
+    props: {
+      decorations(state) {
+        const doc = state.doc;
+
+        if (doc.content.size === 0) {
+          const placeholderDecoration = Decoration.widget(
+            0,
+            () => {
+              const mountNode = document.createElement("span");
+
+              renderPortal(
+                <Typography
+                  component="span"
+                  sx={{
+                    fontSize: "inherit",
+                    color: ({ palette }) => palette.gray[70],
+                  }}
+                >
+                  {text}
+                </Typography>,
+                mountNode,
+              );
+
+              return mountNode;
+            },
+            /**
+             * passing a key prevents a focus related bug, by preventing re-creation of the dom node
+             * @see https://github.com/hashintel/hash/pull/953#issuecomment-1222088538
+             * */
+            {
+              key: "placeholder-deco",
```

Comment:
Not sure if this version needs `key`, I suggest having a look at the behavior. My gut feeling is this is not going to cause an issue for comments.

Even if it is, I'm not sure if we should have the exact same comment in 2 separate places.

Reply:
Removed this in 546418766ecdf7fee2ed340463d3c8d6b6c27dad.